### DISCUSSION TITLE
Nuclei doesn't complete the scan

### DISCUSSION THREAD

**Discussion by smackerdodi** [projectdiscovery/nuclei#1550] [Q&A] [answered]
when i use -stats argument the tool show the number of requests sent and the all number of requests like 525123/1452123 and suddenly stop without complete the scan and send the other requests and the percentage of scan always between 35-80% 
i use it on kali linux vmware with 8 giga ram over host machine of windows 10 
could you please fix this in the next version

**Comment by geeknik** [comment, marked as answer]
This is probably not a bug, I'd encourage you to look at the number of errors during your scan. Nuclei has error handling built in so that if the # of errors per host exceeds a certain threshold, nuclei stops with that host and moves on to the next in your list and so on and so forth. I'd also encourage you to review your command line settings to better understand how you can lower the amount of errors during a scan, for example, don't send so many requests per second. You can also try using the `-trace-log` flag which will give you a better understanding of which hosts are having the problems for example:
```
{
  "id": "CVE-2020-9496",
  "url": "https://origin-apps.tesla.com/webtools/control/xmlrpc",
  "error": "POST https://origin-apps.tesla.com/webtools/control/xmlrpc giving up after 2 attempts: Post \"https://origin-apps.tesla.com/webtools/control/xmlrpc\": read tcp 192.168.1.6:53437->209.133.79.43:443: read: connection reset by peer",
  "type": "http"
}
```

**Comment by MMquant** [comment]
You can disable or tune this error based optimization with these flags

```
OPTIMIZATIONS:

   -mhe, -max-host-error int        max errors for a host before skipping from scan (default 30)
   -nmhe, -no-mhe                   disable skipping host from scan based on errors
```
